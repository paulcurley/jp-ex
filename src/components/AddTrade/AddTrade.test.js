import React from 'react';
import { shallow } from 'enzyme';
import { AddTradeForm } from '.';

const setup = (setupProps = {}) => {
    const props = {
        handleBack: jest.fn(),
        handleAddTrade: jest.fn(),
        ...setupProps,
    };

    const wrapper = shallow(<AddTradeForm {...props} />);

    return {
        props,
        wrapper,
    };
};
describe('Add Trade', () => {
    it('renders', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });
});
