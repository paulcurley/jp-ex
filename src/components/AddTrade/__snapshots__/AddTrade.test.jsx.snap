// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Trade renders 1`] = `
Object {
  "props": Object {
    "handleAddTrade": [MockFunction],
    "handleBack": [MockFunction],
  },
  "wrapper": <form
    onSubmit={[Function]}
  >
    <label>
      Stock Symbol:
      <input
        name="stockSymbol"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </label>
    <label>
      Price:
      <input
        name="price"
        onChange={[Function]}
        required={true}
        type="number"
      />
    </label>
    <label>
      Number of shares:
      <input
        name="noOfShares"
        onChange={[Function]}
        required={true}
        type="number"
      />
    </label>
    <input
      type="submit"
      value="Submit"
    />
  </form>,
}
`;
